### PR TITLE
Allow --insecure option to curl

### PR DESCRIPTION
**Pull request by sathieu**
This can be useful with self-signed certificates, when one don't want (yet) to inject certificate in the pod.

### CODE REVIEW

**Comment by sathieu** [comment]
I've rewritten the code to [make shellcheck happy](https://www.shellcheck.net/wiki/SC2086#exceptions).

File: get-resource.sh [hunk 1]
```diff
@@ -10,10 +10,12 @@ export no_proxy="${no_proxy:-${NO_PROXY:-}}"
...
 SHARED_DIR="${SHARED_DIR:-/shared}"
-CURL_OUTPUT=""
-if [ "${CURL_VERBOSE:-}" = true ]; then
-    CURL_OUTPUT="--verbose"
-fi
+
+curl_with_flags() {
+    [ "${CURL_VERBOSE:-}" != 'true' ] || set -- --verbose "$@"
+    [ "${CURL_INSECURE:-}" != 'true' ] || set -- --insecure "$@"
+    curl "$@"
```
Comment (zaneb) [inline get-resource.sh:17]: What was wrong with

```
if [ "${CURL_INSECURE:-}" = true ]; then
    CURL_OUTPUT="${CURL_OUTPUT} --insecure"
fi
```

?

Comment (sathieu) [reply get-resource.sh:17]: @zaneb shellcheck was [complaining](https://prow.apps.test.metal3.io/view/s3/prow-logs/pr-logs/pull/metal3-io_ironic-ipa-downloader/54/shellcheck/1780915098823430144) for https://github.com/metal3-io/ironic-ipa-downloader/commit/e0bd9e3ca22e77d969167f6bb36b121b457c7e6a.

See https://www.shellcheck.net/wiki/SC2086

**Comment by tuminoid** [comment]
Can we get this merged? It is just changing the syntax and adding second similar option, the proposed implementation is clean IMO.

@zaneb @dtantsur

File: get-resource.sh [hunk 2]
```diff
@@ -10,10 +10,12 @@ export no_proxy="${no_proxy:-${NO_PROXY:-}}"
 
 # configurable variables
 SHARED_DIR="${SHARED_DIR:-/shared}"
-CURL_OUTPUT=""
-if [ "${CURL_VERBOSE:-}" = true ]; then
-    CURL_OUTPUT="--verbose"
-fi
+
+curl_with_flags() {
+    [ "${CURL_VERBOSE:-}" != 'true' ] || set -- --verbose "$@"
```
Comment (dtantsur) [inline get-resource.sh:15]: nit: the double inverted logic caused me a headache

Comment (tuminoid) [reply get-resource.sh:15]: Double inverted is a necessity here. Doing the same with `&&` causes `set -e` to trigger, while `||` catches it and executes the `set -- ...` part.

Comment (sathieu) [reply get-resource.sh:15]:
```bash
[ "${CURL_VERBOSE:-}" = 'true' ] && set -- --verbose "$@" || :
```

may be more readable.

Yes, bash is ugly.

Comment (tuminoid) [reply get-resource.sh:15]: "ternary" operation in Bash is not recommended as it has pitfalls. Here is might work, but better avoid the construct everywhere.

https://mywiki.wooledge.org/BashPitfalls number 22.

Comment (zaneb) [reply get-resource.sh:15]: could we do `if [ "${CURL_VERBOSE:-}" = 'true' ]; then set -- --verbose "$@"; fi` or are we trying to win at code golf? :slightly_smiling_face:

Comment (sathieu) [reply get-resource.sh]: @zaneb I pushed a more readable version, following your advice.

Back to you!

**Comment by sathieu** [comment]
@dtantsur The CI is failing, but this doesn't looks related to this PR. Any idea?

**Comment by tuminoid** [comment]
> /test metal3-centos-e2e-integration-test-main
> 
> Let's give it one more shot

This is expected to fail since CAPI e2e code is broken until tonights release. :(

**Pull request merged**